lib/components/mobile/main.js

```diff
@@ -69,8 +71,13 @@ class MobileMain extends Component {
         )
 
       case MobileScreens.SEARCH_FORM:
+        // Render batch search screen if batch routing enabled. Otherwise,
+        // default to standard search screen.
+        const SearchScreen = isBatchRoutingEnabled(config)
+          ? BatchSearchScreen
```

Comment:
This kind of goes against the idea that the implementing application is able to choose between either the DefaultMainPanel and the BatchRoutingPanel. In here, the BatchSearchScreen and all its associated dependencies are always imported even if they aren't supposed to be used. Perhaps the SearchScreen can be made into a required prop that must be set by the implementing app.

Reply:
Yea, agreed. I had this thought also. Thanks for flagging. I was considering whether SearchScreen should be a required prop or if we should have `mobileView` be configurable in trimet-mod-otp and have a separate file from `main.js` for batch routing that handles the functionality/logic from this file. Right now, there's only minor differences between the two (the search screen), but the option to define different `mobileViews` in trimet-mod-otp might be nice to have. Thoughts?

Reply:
After looking at this more, I'm not sure why there are even slots for `desktopView` and `mobileView` at all, especially now that we're allowing the configuration of custom components via the context API. The example.js and trimet-mod-otp files are nearly identical in how they render all this. Also, all the stuff in `desktopView` and `mobileView` is super connected to all the otp-react-redux internals that I don't think you'd ever want to define a cust mobileView and if you did, you'd just be writing a different app anyways.

I've gone ahead and refactored `ResponsiveWebapp` so that it does away with the `desktopView` and `mobileView` slots completely and now it relies more on the Context API for the things we did want to customize. See https://github.com/opentripplanner/otp-react-redux/pull/337.